gtk4/src/actionable.rs

```diff
@@ -1,25 +1,22 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
 use crate::Actionable;
-use glib::object::IsA;
 use glib::translate::*;
-use libc::c_char;
+use glib::{IsA, ToVariant};
 
 // rustdoc-stripper-ignore-next
 /// Trait containing manually implemented methods of [`Actionable`](crate::Actionable).
 pub trait ActionableExtManual: 'static {
     #[doc(alias = "gtk_actionable_set_action_target")]
-    fn set_action_target(&self, string: &str);
+    fn set_action_target<V: ToVariant>(&self, string: &V);
 }
 
 impl<O: IsA<Actionable>> ActionableExtManual for O {
-    fn set_action_target(&self, string: &str) {
-        let string: Stash<*const c_char, _> = string.to_glib_none();
+    fn set_action_target<V: ToVariant>(&self, string: &V) {
         unsafe {
             ffi::gtk_actionable_set_action_target(
                 self.as_ref().to_glib_none().0,
-                b"%s\0".as_ptr() as *const c_char,
-                string.0,
+                string.to_variant().type_().as_str().to_glib_none().0,
```

Comment:
... and can do just

```suggestion
                V::static_variant_type.to_glib_none().0,
```

Reply:
However I think this is wrong. This function wants a variant format string and then the values of the variant.

I think you want to call `gtk_actionable_set_action_target_value(v.to_variant().to_glib_none().0)`. Note that this function's parameter is `nullable`.